service/tracker/initialize.go

```diff
@@ -18,24 +18,22 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"path/filepath"
+	"io"
 
 	"github.com/dgraph-io/badger/v2"
 
-	"github.com/onflow/flow-go/model/bootstrap"
 	"github.com/onflow/flow-go/module/metrics"
 	protocol "github.com/onflow/flow-go/state/protocol/badger"
 	"github.com/onflow/flow-go/state/protocol/inmem"
 	"github.com/onflow/flow-go/storage"
 	cache "github.com/onflow/flow-go/storage/badger"
 	"github.com/onflow/flow-go/storage/badger/operation"
-	"github.com/onflow/flow-go/utils/io"
 )
 
 // Initialize will initialize the Flow protocol state in the given database. The
 // code is inspired by the related unexported code in the Flow Go code base:
 // https://github.com/onflow/flow-go/blob/v0.21.0/cmd/bootstrap/cmd/finalize.go#L452
-func Initialize(dir string, db *badger.DB) error {
+func Initialize(reader io.Reader, db *badger.DB) error {
```

Comment:
OK, I see now why you changed this. I take my comment back, this is probably better than, for testing / loose coupling reasons. Let's open files in main for everything, for the same reason.

Reply:
We can always create temporary files in the tests, but what I don't like with this (on top of the fact that I don't like tests making changes on the machine) is that it's often unreliable in CIs and can be problematic in systems that do not clean up their `/tmp` (Hello WSL!).

So I'm glad that you changed your mind seeing this 👍